File: src/main/java/com/optica/util/UserValidator.java
package com.optica.util;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;
import org.springframework.validation.Errors;
import org.springframework.validation.ValidationUtils;
import org.springframework.validation.Validator;

import com.optica.entity.User;
import com.optica.services.login.UserService;

@Component
public class UserValidator implements Validator {

	@Autowired
	private UserService userService;

	@Override
	public boolean supports(Class<?> arg0) {
		return User.class.equals(arg0);
	}

	@Override
	public void validate(Object o, Errors errors) {
		User user = (User) o;

		ValidationUtils.rejectIfEmptyOrWhitespace(errors, "username", "NotEmpty");
		if (user.getUsername().length() < 2 || user.getUsername().length() > 32) {
			errors.rejectValue("username", "Size");
		}
		if (userService.findByUsername(user.getUsername()) != null) {
			errors.rejectValue("username", "Duplicate");
		}

		ValidationUtils.rejectIfEmptyOrWhitespace(errors, "password", "NotEmpty");
		if (user.getPassword().length() < 2 || user.getPassword().length() > 32) {
			errors.rejectValue("password", "Size");
		}

	}

}

File: src/main/kotlin/com/optica/util/UserValidator.kt
package com.optica.util

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component
import org.springframework.validation.Errors
import org.springframework.validation.ValidationUtils
import org.springframework.validation.Validator

import com.optica.entity.User
import com.optica.services.login.UserService

@Component
class UserValidator : Validator {

    @Autowired
    private val userService: UserService? = null

    override fun supports(arg0: Class<*>): Boolean {
        return User::class.java == arg0
    }

    override fun validate(o: Any?, errors: Errors) {
        val user = o as User?

        ValidationUtils.rejectIfEmptyOrWhitespace(errors, "username", "NotEmpty")
        if (user!!.username!!.length < 2 || user.username!!.length > 32) {
            errors.rejectValue("username", "Size")
        }
        if (userService!!.findByUsername(user.username) != null) {
            errors.rejectValue("username", "Duplicate")
        }

        ValidationUtils.rejectIfEmptyOrWhitespace(errors, "password", "NotEmpty")
        if (user.password!!.length < 2 || user.password!!.length > 32) {
            errors.rejectValue("password", "Size")
        }

    }

}
